File: Water Bottles/water_bottles.cpp
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        int res = numBottles ; 
        while(numBottles/numExchange >= 1){
            res = res + numBottles/numExchange;
            numBottles = numBottles/numExchange + numBottles%numExchange;
        }
        return res ; 
        
    }
};

File: Water Bottles/water_bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        res = numBottles
        while numBottles // numExchange >= 1 :
            res+=numBottles // numExchange
            numBottles = numBottles // numExchange + numBottles % numExchange
        return res